### PR TITLE
Closes AgileVentures/Metplus_tracker#374 

### PR DESCRIPTION
**Pull request by sandhyakoti**

### CODE REVIEW

File: app/views/jobs/_job_applications.html.haml [hunk 1]
```diff
@@ -29,18 +29,25 @@
                   = job_application.job_seeker.job_seeker_status.short_description
               - when :updated_at
                 %td
-                  - applied_at = job_application.status_change_time(:active)
-                  = "#{time_ago_in_words(applied_at, )} ago"
+                  = "#{time_ago_in_words(job_application.updated_at, )} ago"
```
Comment (patmbolger) [inline app/views/jobs/_job_applications.html.haml]: The two deleted statements need to be restored.

File: spec/models/event_spec.rb [hunk 2]
```diff
@@ -6,14 +6,12 @@
   let!(:agency)        { FactoryGirl.create(:agency) }
   let(:agency_admin)   { FactoryGirl.create(:agency_admin) }
   let!(:job_developer)  { FactoryGirl.create(:job_developer, agency: agency) }
-  let!(:case_manager)   { FactoryGirl.create(:case_manager, agency: agency) }
+  let!(:case_manager)   { FactoryGirl.create(:case_manager) }
```
Comment (patmbolger) [inline spec/models/event_spec.rb]: The deleted line needs to be restored.

It is associated with PR #385

File: spec/models/event_spec.rb [hunk 3]
```diff
@@ -6,14 +6,12 @@
...
   let!(:job_seeker) do
     js = FactoryGirl.create(:job_seeker)
     js.assign_job_developer(job_developer, agency)
-    js.assign_case_manager(case_manager, agency)
```
Comment (patmbolger) [inline spec/models/event_spec.rb]: Same comment as above.

File: spec/models/event_spec.rb [hunk 4]
```diff
@@ -116,7 +114,7 @@
                          {job_id:  job.id,
                           js_id:   job_seeker.id,
                           js_name: job_seeker.full_name(last_name_first: false),
-                          notify_list: [job_seeker.case_manager.user.id, job_seeker.job_developer.user.id, company_person.user.id]})
```
Comment (patmbolger) [inline spec/models/event_spec.rb]: Same comment as above.

File: spec/models/event_spec.rb [hunk 5]
```diff
@@ -203,30 +202,42 @@
...
-    end
-
-    it 'triggers Pusher message to case manager' do
-      Event.create(:APP_ACCEPTED, application)
-      expect(Pusher).to have_received(:trigger).
-                        with('pusher_control',
-                             'job_application_accepted',
-                             {id: application.id,
-                              ap_user_id: application.job_seeker.case_manager.user.id,
+                              jd_user_id: application.job_seeker.job_developer.user.id,
```
Comment (patmbolger) [inline spec/models/event_spec.rb]: Same comment as above.

File: spec/models/event_spec.rb [hunk 6]
```diff
@@ -203,30 +202,42 @@
...
                               job_title: job.title,
                               js_name: application.job_seeker.full_name(last_name_first: false)
                               })
     end
-    
-    it 'sends a notification email to job developer and case manager' do
```
Comment (patmbolger) [inline spec/models/event_spec.rb]: Same comment as above.

File: app/assets/javascripts/job_applications.js [hunk 7]
```diff
@@ -1,90 +1,185 @@
...
+
+            $(action).find('#confirm_reject_noxhr').attr('action',
+                '/job_applications/' + id + '/reject');
+
+            $('#job_reject_errors').hide(); // Hide error div in modal
+        }
+        $(action).modal();
+        event.preventDefault();
+    });
+});
```
Comment (sandhyakoti) [inline app/assets/javascripts/job_applications.js]: I didn't get this. Where the lines need to be moved?

Comment (patmbolger) [inline app/assets/javascripts/job_applications.js]: Looking at the file as it exists now, lines *85 - 102* should be moved to immediately after line 82, and before line 83.

That will cause them to be included in the function that is called when the page is loaded.

File: features/step_definitions/fixtures_steps.rb [hunk 8]
```diff
@@ -191,11 +191,12 @@
     end
 
     job.save!
-
+    puts "jobe creation - done" + job.title
     if skills and not skills.empty?
       skills.split(/(?:,\s*)/).each do |skill|
         JobSkill.create(job: job, skill: Skill.find_by_name(skill),
                         required: true, min_years: 1, max_years: 20)
+        puts "skill inserted:" + skill
```
Comment (patmbolger) [inline features/step_definitions/fixtures_steps.rb]: This looks like a hold-over from development (?)

**Comment by patmbolger** [comment]
PR looks good to go.   

Thanks for persevering on this challenging story.

**Comment by patmbolger** [comment]
Sandy - I had that indentation problem too.  Your editor is using tabs, and I suspect the setting for tabs in the editor is 2 spaces.  However, the github spacing is 8 spaces - hence the wider indentation on github.

You should be able to set your editor to insert spaces, instead of a tab, when the tab button is pressed.

**Pull request merged**